trino/sqlalchemy/dialect.py

```diff
@@ -280,7 +280,7 @@ def get_indexes(self, connection: Connection, table_name: str, schema: str = Non
         if not self.has_table(connection, table_name, schema):
             raise exc.NoSuchTableError(f"schema={schema}, table={table_name}")
 
-        partitioned_columns = self._get_columns(connection, f"{table_name}$partitions", schema, **kw)
+        partitioned_columns = self._get_columns(connection, table_name, schema, **kw)
```

Comment:
`$partitions` table depends on the connector's implementation, so we shouldn't have used it. 
We could change this method like a) change the behavior based on the connector name or b) keep `$partitions` and make it exception driven. 

Removing `$partitions` table suffix doesn't make sense to me. 

cc: @hashhar

Reply:
thank you for sharing your opinion! We're using the Hudi connector so if I choose the A pattern, this should be fixed like as follows, right...? (please feel free to make commits to this branch or create another PR 🙇 )
```python
 if the connector is hudi:
    self._get_columns(connection, table_name, schema, **kw)
 else:
   partitioned_columns = self._get_columns(connection, table_name, schema, **kw)
```

Reply:
@ebyhr thank you for your comment!

>  but we can start with b) in my opinion.

Following this, I made a commit https://github.com/trinodb/trino-python-client/pull/426/commits/2232541316bd31ea89c395ce3dd781381fac5861

Could you check if this is what you expect...? 🙏 

---
>  e.g. parse result of SHOW CREATE TABLE,

sorry, I don't understand how to parse this result. 🤦 

Here is the result of `SHOW CREATE TABLE`  for the target Hudi table:

- SHOW CREATE TABLE <TABLE> using Trino
```
Create Table
CREATE TABLE "<CATALOG>"."<DB>"."<TABLE>" (
   _hoodie_commit_time varchar COMMENT '',
   _hoodie_commit_seqno varchar COMMENT '',
   _hoodie_record_key varchar COMMENT '',
   _hoodie_partition_path varchar COMMENT '',
   _hoodie_file_name varchar COMMENT '',
   global_intensity double COMMENT '',
   global_reactive_power double COMMENT '',
   city varchar COMMENT '',
   voltage double COMMENT '',
   global_active_power double COMMENT '',
   sub_metering_1 double COMMENT '',
   sub_metering_2 double COMMENT '',
   sub_metering_3 double COMMENT '',
   meter_id varchar COMMENT '',
   location array(double) COMMENT '',
   ts varchar COMMENT ''
)
WITH (
   location = 's3a://<my-bucket>',
   partitioned_by = ARRAY['ts']
)
```

-  SHOW CREATE TABLE <TABLE> using Presto
```
Create Table
CREATE TABLE hudi."data-platform-demo"."hudi-s3-ingest " (
   "_hoodie_commit_time" varchar,
   "_hoodie_commit_seqno" varchar,
   "_hoodie_record_key" varchar,
   "_hoodie_partition_path" varchar,
   "_hoodie_file_name" varchar,
   "global_intensity" double,
   "global_reactive_power" double,
   "city" varchar,
   "voltage" double,
   "global_active_power" double,
   "sub_metering_1" double,
   "sub_metering_2" double,
   "sub_metering_3" double,
   "meter_id" varchar,
   "location" array(double),
   "ts" varchar
)
```

The difference is that Trino has:
```
WITH (
   location = 's3a://<my-bucket>',
   partitioned_by = ARRAY['ts']
)
```

In Superset, fetching Hudi schema works in Presto but it does not work in Trino.
How can we use this information...? 🙇

Reply:
I think Yuya meant that instead of using `$partitions` table in case of Hudi you can fire a `SHOW CREATE TABLE` and use the `partitioned_by` information from that output instead.

IMO however the simple fix is to do something like:

```
        partitioned_columns = None
        try:
            partitioned_columns = self._get_columns(connection, f"{table_name}$partitions", schema, **kw)
        except Exception as e:
            logger.debug("Couldn't fetch partition columns for ...")
        if not partitioned_columns:
            return []
        partition_index = dict(
            name="partition",
            column_names=[col["name"] for col in partitioned_columns],
            unique=False
        )
        return [partition_index]
```

This feature shouldn't have been added to begin with since there's no general purpose way to figure out if a table is partitioned or not at the moment.

Reply:
@hashhar thank you for your comment! I fixed the code to follow your implementation.
I see... Could we use the simple fix for now to make Superset work...?  🙏 
When creating the dataset in Superset using Trino+Hudi, this fetching error blocks it. There is a workaround to create the dataset via `SQL Lab` but we want to use the normal way to create the dataset.